artifact/artifact.sh

```diff
@@ -1,48 +1,89 @@
 #!/bin/bash -eux
-#
-# This code is meant to be executed within a TravisCI build.
+
+# This code is meant to be executed within a CI build.
 
 publish_to_dockerhub() {
     if [ -n "${DOCKER_USERNAME:-}" ] && [ -n "${DOCKER_PASSWORD:-}"  ]; then
-        DOCKER_REPO="tidepool/${TRAVIS_REPO_SLUG#*/}"
-        echo "${DOCKER_PASSWORD}" | docker login --username "${DOCKER_USERNAME}" --password-stdin
 
-        if [ "${TRAVIS_REPO_SLUG:-}" == "tidepool-org/blip" ]; then
+        # Determine CI provider
+        if [ -n "${TRAVIS:-}" ]; then
+            CI_PROVIDER="travis"
+        elif [ -n "${CIRCLECI:-}" ]; then
+            CI_PROVIDER="circle"
+        else
+            echo "No known CI provider detected"
+            return 1
+        fi
+
+        # Set common variables
+        if [ "$CI_PROVIDER" = "travis" ]; then
+            REPO_SLUG=$TRAVIS_REPO_SLUG
+            COMMIT=$TRAVIS_COMMIT
+            TAG=${TRAVIS_TAG:-}
+            BRANCH=${TRAVIS_BRANCH:-}
+            PR_BRANCH=${TRAVIS_PULL_REQUEST_BRANCH:-}
+        elif [ "$CI_PROVIDER" = "circle" ]; then
+            REPO_SLUG="${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}"
+            COMMIT=$CIRCLE_SHA1
+            TAG=${CIRCLE_TAG:-}
+            BRANCH=${CIRCLE_BRANCH:-}
+            PR_BRANCH=${CIRCLE_PR_NUMBER:-}
```

Comment:
This is going to yield two different results whether it was run in Travis or Circle. In the Travis case, the Docker image tag will be based upon the feature branch, while in Circle it will be based upon the PR number.

Note, this is already under discussion in https://tidepoolteam.slack.com/archives/CF59ADWLB/p1699578208270199.

Reply:
Personally I think we should use the PR number (from both Travis and Circle) to tag the PR build images. That is currently under discussion, though. I'd prefer to wait until that is resolved and update your PR with that change (if any) so we only do the work once.

Reply:
Yeah, I had to work around the discrepancies between what ENV variables are available on Circle vs Travis. Travis doesn't have PR number (though they do have PR SHA). https://docs.travis-ci.com/user/environment-variables/#default-environment-variables vs https://circleci.com/docs/variables/#built-in-environment-variables

Reply:
What about `TRAVIS_PULL_REQUEST`? According to the docs - "The pull request number if the current job is a pull request, “false” if it’s not a pull request."

Reply:
That should work 👍🏼